authn/keychain.go

```diff
@@ -41,14 +42,23 @@ func configDir() (string, error) {
 	if dc := os.Getenv("DOCKER_CONFIG"); dc != "" {
 		return dc, nil
 	}
-	if h := os.Getenv("HOME"); h != "" {
-		return path.Join(h, ".docker"), nil
+	if h := homeDir(); h != "" {
+		return path.Join(homeDir(), ".docker"), nil
 	}
-	usr, err := user.Current()
-	if err != nil {
-		return "", err
+	return "", errNoHomeDir
+}
+
+var errNoHomeDir = errors.New("could not determine home directory")
+
+func homeDir() string {
+	if runtime.GOOS == "windows" {
+		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
```

Comment:
Left comments in the previous thread. tl;dr Docker specifically uses %USERPROFILE% on Windows

Reply:
Okay, the new path is DOCKER_CONFIG -> USERPROFILE -> HOME, regardless of platform.

If this is terrible we can change it in the future.